[GlobalISel][AArch64] Allow vector ptr to int unmerges

davemgreen opened the pull request:
Vector pointer -> scalar integer unmerges are already legal. This loosens the verifier check for vector-of-pointers -> vectors.

arsenm commented on llvm/test/MachineVerifier/test_g_unmerge_values.mir:
```diff
@@ -30,4 +30,9 @@ body: |
     ; CHECK: Bad machine code: G_UNMERGE_VALUES scalar source operand does not match scalar destination operands
     %17:_(s32), %18:_(s32), %19:_(s32) = G_UNMERGE_VALUES %14
+
+    ; CHECK-NOT: Bad machine code
+    %20:_(<4 x p0>) = IMPLICIT_DEF
+    %21:_(<2 x s64>), %22:_(<2 x s64>) = G_UNMERGE_VALUES %20
+    %23:_(s64), %24:_(s64), %25:_(s64), %26:_(s64) = G_UNMERGE_VALUES %20
```
Positive checks for the verifier should not go in the same file as the error cases, the not check is fragile

davemgreen replied:
Is the getelementptr.ll test enough for you? It runs into verifier failures without this change.

davemgreen commented:
I've switched over to a irtranslator test - that hopefully more clearly tests this including where they get generated.

arsenm commented on llvm/test/CodeGen/AArch64/irtranslator-ptrvec.ll:
```diff
@@ -0,0 +1,23 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+; RUN: llc -O0 -mtriple=aarch64-linux-gnu -global-isel -verify-machineinstrs -stop-after=irtranslator %s -o - | FileCheck %s
+
+define <4 x ptr> @vector_gep_v4i32(<4 x ptr> %b, <4 x i32> %off) {
```
Can just merge this into translate-gep.ll

arsenm reviewed (approved):
lgtm with nit

davemgreen committed:
[GlobalISel][AArch64] Allow vector ptr to int unmerges

Vector pointer -> scalar integer unmerges are already legal. This loosens the
verifier check for vector-of-pointers -> vectors.
